src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -56,7 +58,7 @@
 
 public class GameParser {
   private static final Class<?>[] SETTER_ARGS = {String.class};
-  private GameData data;
+  private final GameData data = new GameData();
```

Comment:
Although this slightly changes functionality, it's safe to change it, because all callers create fresh instance of GameParser

Reply:
Agree it's safe.  As I mentioned elsewhere, `GameParser` appears to be a _Method Object_, and thus should only ever be "used" once.  It might be useful to update the class Javadocs stating that once one of the public `parseXXX()` methods is called, the object can no longer be used.  Some example code might explain that nicely.

An alternative is to just make it impossible to make that mistake.  You could make the constructor and all instance methods private and just provide public static methods that handle the two use cases.  For example:

```java
public static GameData parse(final String mapName, final InputStream stream, final AtomicReference<String> gameName)
    throws GameParseException, SAXException, EngineVersionException, IllegalArgumentException {
  return new GameParser(mapName).parse(stream, gameName);
}
```
```